logagg_master/cli.py

```diff
@@ -0,0 +1,417 @@
+import sys
+from os.path import expanduser
+
+import ujson as json
+from tabulate import tabulate
+import requests
+from diskdict import DiskDict
+from logagg_utils import ensure_dir
+from deeputil import AttrDict
+from structlog.dev import ConsoleRenderer
+
+def prGreen(txt): print("\033[92m {}\033[00m" .format(txt))
+def prRed(err): print("\033[91m {}\033[00m" .format(err))
+def prYellow(txt): print("\033[93m {}\033[00m" .format(txt))
+
+class LogaggCli():
+    '''
+    Command line interface for logagg
+    '''
+    
+    MASTER_PING_URL = 'http://{host}:{port}/logagg/v1/ping?key={key}&secret={secret}'
+    MASTER_ADD_NSQ_URL = 'http://{host}:{port}/logagg/v1/add_nsq?nsqd_tcp_address={nsqd_tcp_address}&nsqd_http_address={nsqd_http_address}&key={key}&secret={secret}'
+    MASTER_GET_NSQ_URL = 'http://{host}:{port}/logagg/v1/get_nsq?key={key}&secret={secret}'
+    GET_TOPIC_URL = 'http://{host}:{port}/logagg/v1/get_topics'
+    GET_TOPIC_INFO_URL = 'http://{host}:{port}/logagg/v1/get_topic_info?topic_name={topic_name}'
+    GET_COMPONENT_URL = 'http://{host}:{port}/logagg/v1/get_components?topic_name={topic_name}'
+    TAIL_LOGS_URL = 'http://{host}:{port}/logagg/v1/tail_logs?topic_name={topic_name}'
+    COLLECTOR_ADD_FILE_URL = 'http://{host}:{port}/logagg/v1/collector_add_file?topic_name={topic_name}&collector_host={collector_host}&collector_port={collector_port}&fpath="{fpath}"&formatter="{formatter}"'
+    COLLECTOR_REMOVE_FILE_URL = 'http://{host}:{port}/logagg/v1/collector_remove_file?topic_name={topic_name}&collector_host={collector_host}&collector_port={collector_port}&fpath="{fpath}"'
+
+    def __init__(self):
+        self.data_path = ensure_dir(expanduser('~/.logagg'))
+        self.state = DiskDict(self.data_path)
+        self._init_state()
+
+
+    def _init_state(self):
+        '''
+        Initialize default values for stored state
+        '''
+        if not self.state['master']:
```

Comment:
Should throw an error in case of normal dictionaries but diskdicts work differently
![image](https://user-images.githubusercontent.com/33823698/48847089-b6bdce00-edc6-11e8-9b7a-c4e08cbfaf75.png)